src/lighteval/tasks/templates/multichoice.py

```diff
@@ -0,0 +1,130 @@
+# MIT License
+
+# Copyright (c) 2024 The HuggingFace Team
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from typing import Callable
+
+from typing_extensions import NotRequired, TypedDict
+
+from lighteval.tasks.requests import Doc
+from lighteval.tasks.templates.utils.formatting_utils import capitalize, fix_ending_punct
+from lighteval.tasks.templates.utils.formulation import Formulation, MCFFormulation, build_answers, build_options
+from lighteval.tasks.templates.utils.translation_literals import TRANSLATION_LITERALS
+from lighteval.tasks.templates.utils.utils import create_adapter_from_dict
+from lighteval.utils.language import Language
+from lighteval.utils.utils import as_list
+
+
+MULTI_CHOICE_QA_QUERY = (
+    "{instruction}{context}{question_word}{colon}{sentence_space}{question}\n{options}{answer_word}{colon}"
+)
+
+
+# NO idea how to ensure we have the same keys in these typedicts in python :(
+class MCQInput(TypedDict):
+    question: str
+    choices: list[str]
+    gold_idx: list[int] | int
+    context: NotRequired[str]
+    instruction: NotRequired[str]
+
+
+class MCQDictAdapter(TypedDict):
+    question: str
+    choices: str
+    gold_idx: str
+    context: NotRequired[str]
+    instruction: NotRequired[str]
```

Comment:
I still don't see what these are used for apart from indicating the schema that should be followed, and you could use doc for this instead of adding new classes